pgoapi/pgoapi.py

```diff
@@ -627,39 +640,65 @@ def cleanup_pokemon(self, inventory_items=None):
         # never release favorites and other defined pokemons
         if pokemon.is_favorite or pokemon.pokemon_id in self.keep_pokemon_ids:
-            return False
-        elif self.RELEASE_DUPLICATES and (
-                    self.pokemon_lvl(best_pokemon) * self.RELEASE_DUPLICATES_SCALER > self.pokemon_lvl(
-                    pokemon) and self.pokemon_lvl(pokemon) < self.RELEASE_DUPLICATES_MAX_LV):
-            return True
-        # release defined throwaway pokemons  but make sure we have kept at least 1 (dont throw away all of them)
-        elif pokemon.pokemon_id in self.throw_pokemon_ids:
-            return True
+            return False, False
+        # release defined throwaway pokemons, but make sure we have kept at least 1 (dont throw away all of them)
+        elif pokemon.pokemon_id in self.throw_pokemon_ids and pokemons_keep_counter >= self.MIN_SIMILAR_POKEMON:
+            return True, False
+        # release duplicates if applicable
+        elif self.RELEASE_METHOD == "DUPLICATES":
+            if (self.pokemon_score(best_pokemon) * self.RELEASE_DUPLICATES_SCALAR > self.pokemon_score(pokemon) and
+                    self.pokemon_score(pokemon) < self.RELEASE_DUPLICATES_MAX_SCORE) and \
+                            pokemons_keep_counter >= self.MIN_SIMILAR_POKEMON:
+                return True, False
+            else:
+                return False, False
+        # keep high-iv pokemons based on config values
+        elif pokemon.iv_normalized > self.KEEP_IV_OVER \
+                and pokemons_keep_iv_counter < self.MAX_POKEMON_HIGH_IV \
+                and pokemon.cp > (best_pokemon.cp * self.KEEP_IV_WITH_PERCENT_CP / 100):
+            return False, True
         # keep high-cp pokemons
-        elif pokemon.cp > self.KEEP_CP_OVER or pokemon.iv > self.KEEP_IV_OVER:
-            return False
-        # if we haven't found a reason to keep it, transfer it
-        else:
-            return True
-
-    def pokemon_lvl(self, pokemon):
-        if self.DEFINE_POKEMON_LV == "CP":
+        elif pokemon.cp > self.KEEP_CP_OVER:
+            return False, False
+        # release all other pokemons
+        elif pokemons_keep_counter >= self.MIN_SIMILAR_POKEMON:
+            return True, False
+        # if we haven't found a reason, keep it! (probably not at MIN_SIMILAR yet)
+        return False, False
+
+    def pokemon_score(self, pokemon):
+        if self.SCORE_METHOD == "CP":
             return pokemon.cp
-        elif self.DEFINE_POKEMON_LV == "IV":
-            return pokemon.iv
-        elif self.DEFINE_POKEMON_LV == "CP*IV":
-            return pokemon.cp * pokemon.iv
-        elif self.DEFINE_POKEMON_LV == "CP+IV":
-            return pokemon.cp + pokemon.iv
+        elif self.SCORE_METHOD == "IV":
+            return pokemon.iv_normalized
+        elif self.SCORE_METHOD == "CP*IV":
+            return pokemon.cp * pokemon.iv_normalized
+        elif self.SCORE_METHOD == "CP+IV":
```

Comment:
I still believe we should remove CP*IV and CP+IV as options since you're literally multiplying/adding a potentially very large number with a number between 0-100. I can't even comprehend what that means and if someone were to use these methods, it could release a lot of pokemon that's not intended.

I think just the "CP", "IV", and "FANCY" methods would be sufficient and cover most cases, and are more intuitive.